ci/functional/node_checkout.sh

```diff
@@ -0,0 +1,211 @@
+#!/bin/bash
+
+# Hardware Validation script to verify if the single node
+# is in deployable state. This is intended to run on a single node.
+
+rc=1
+
+# Verify if all the required rpms are installed
+rpms_check(){
+
+prereq_rpms_list=("boost-python36-devel" "clang-analyzer" "cmake" "CUnit-devel" "doxygen" "e2fsprogs" "file" "flex" "fuse3-devel" "gcc" "gcc-c++" "git" "golang" "graphviz" "hwloc-devel" "ipmctl" "java-1.8.0-openjdk" "json-c-devel" "lcov" "libaio-devel" "libcmocka-devel" "libevent-devel" "libipmctl-devel" "libiscsi-devel" "libtool" "libtool-ltdl-devel" "libunwind-devel" "libuuid-devel" "libyaml-devel" "Lmod" "lz4-devel" "make" "man" "maven" "nasm" "ndctl" "numactl" "numactl-devel" "openmpi3-devel" "openssl-devel" "pandoc" "patch" "patchelf" "pciutils" "python36-Cython" "python36-devel" "python36-distro" "python36-junit_xml" "python36-numpy" "python36-paramiko" "python36-pylint" "python36-requests" "python36-tabulate" "python36-pyxattr" "python36-PyYAML" "python36-scons" "sg3_utils" "sudo" "valgrind-devel" "yasm")
+
+daos_rpms_servers=("argobots" "daos" "daos-server" "dpdk" "libfabric" "libpmem" "libpmemobj" "mercury" "protobuf-c" "spdk" "spdk-tools")
+
+daos_rpms_clients=("daos-client" "daos" "libfabric" "mercury" "protobuf-c")
+
+if [[ $1 == "server" ]]; then
+    rpms_list=("${prereq_rpms_list[@]}" "${daos_rpms_servers[@]}")
+elif [[ $1 == "client" ]]; then
+    rpms_list=("${prereq_rpms_list[@]}" "${daos_rpms_clients[@]}")
+else
+    rpms_list=("${prereq_rpms_list[@]}")
+fi
+
+echo "Checking for following list of rpms: ${rpms_list[@]}"
+
+declare -a rpms_installed
+declare -a rpms_not_installed
+
+for i in "${!rpms_list[@]}"; do
+    rc_rpm=`rpm -qa | grep -q ${rpms_list[i]};echo $?`
+    if [ "${rc_rpm}" != 0 ]; then
+        rpms_not_installed=(${rpms_not_installed[@]} "${rpms_list[i]}")
+    else
+        rpms_installed=(${rpms_installed[@]} "${rpms_list[i]}")
+    fi
+done
+
+echo "Installed rpms: ${rpms_installed[@]}"
+
+if [ ${#rpms_not_installed[@]} != 0 ]; then
+    echo "[FAILED] ${rpms_not_installed[@]} not installed"
+    exit $rc
+fi
+
+}
+
+# Verify with lscpu that VT-x is enabled
+vtx_enabled(){
+
+output=`lscpu | grep Virtualization`
+awk '$0~/VT-x/{print "VT-x is enabled\n"}' <<< $output
+
+}
+
+# Verify that the number of NUMA nodes is equal to the number of sockets
+check_numa_sockets(){
+
+output=`lscpu | egrep -e Socket -e NUMA`
```

Comment:
(lint) Use $(...) notation instead of legacy backticked `...`. [SC2006]
(lint) egrep is non-standard and deprecated. Use grep -E instead. [SC2196]